https://github.com/dateutil/dateutil/commit/88ee381e2bb1f19da22d252cf68a2d02016d7df3.patch

From 88ee381e2bb1f19da22d252cf68a2d02016d7df3 Mon Sep 17 00:00:00 2001
From: Paul Ganssle <git@m.ganssle.io>
Date: Sat, 26 Sep 2026 12:20:01 -0400
Subject: [PATCH] Pass a list rather than a generator to parametrize

pytest 9.1 deprecated passing a non-Collection iterable to
`pytest.mark.parametrize` and emits `PytestRemovedIn10Warning` when it
sees one. Since we run with `filterwarnings = error`, the warning turns
into a collection error for `tests/test_isoparser.py`, which takes down
every job that installs a current pytest.

The surrounding decorators were reformatted by the darker pre-commit
hook.
---
 changelog.d/1577.misc.rst |  1 +
 tests/test_isoparser.py   | 15 +++++++++------
 2 files changed, 10 insertions(+), 6 deletions(-)
 create mode 100644 changelog.d/1577.misc.rst

diff --git changelog.d/1577.misc.rst changelog.d/1577.misc.rst
new file mode 100644
index 000000000..2738b87db
--- /dev/null
+++ changelog.d/1577.misc.rst
@@ -0,0 +1 @@
+Fixed collection of ``tests/test_isoparser.py`` under pytest 9.1, which turns passing a generator to ``parametrize`` into an error when combined with ``filterwarnings = error``. Reported by @apoorvdarshan (gh pr #1531) and @Osamaali313 (gh pr #1558). (gh pr #1577)
diff --git tests/test_isoparser.py tests/test_isoparser.py
index c33881f3b..b284a6c85 100644
--- tests/test_isoparser.py
+++ tests/test_isoparser.py
@@ -118,12 +118,15 @@ def test_ymd_hms(dt, date_fmt, time_fmt, tzoffset):
     _isoparse_date_and_time(dt, date_fmt, time_fmt, tzoffset)
 
 DATETIMES = [datetime(2017, 11, 27, 6, 14, 30, 123456)]
-@pytest.mark.parametrize('dt', tuple(DATETIMES))
-@pytest.mark.parametrize('date_fmt', YMD_FMTS)
-@pytest.mark.parametrize('time_fmt', (x + sep + '%f' for x in HMS_FMTS
-                                      for sep in '.,'))
-@pytest.mark.parametrize('tzoffset', TZOFFSETS)
-@pytest.mark.parametrize('precision', list(range(3, 7)))
+
+
+@pytest.mark.parametrize("dt", tuple(DATETIMES))
+@pytest.mark.parametrize("date_fmt", YMD_FMTS)
+@pytest.mark.parametrize(
+    "time_fmt", [x + sep + "%f" for x in HMS_FMTS for sep in ".,"]
+)
+@pytest.mark.parametrize("tzoffset", TZOFFSETS)
+@pytest.mark.parametrize("precision", list(range(3, 7)))
 def test_ymd_hms_micro(dt, date_fmt, time_fmt, tzoffset, precision):
     # Truncate the microseconds to the desired precision for the representation
     dt = dt.replace(microsecond=int(round(dt.microsecond, precision-6)))
